File: LeetCode/Easy/0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        char[] ch = s.toLowerCase().replaceAll("[^a-z0-9]", "").toCharArray();
        int len = ch.length;
        int mid = len / 2;
        
        for (int i = 0; i < mid; i++) {
            if (ch[i] != ch[len - 1 - i]) {
                return false;
            }
        }
        return true;
    }
}

File: LeetCode/Easy/0125-valid-palindrome/0125-valid-palindrome.kt
class Solution {
    fun isPalindrome(s: String): Boolean {
        val ch = s.lowercase().replace(Regex("[^a-z0-9]"), "").toCharArray()
        val len = ch.size
        val mid = len / 2

        for (i in 0..< mid) {
            if (ch[i] != ch[len - 1 - i]) {
                return false
            }
        }
        return true
    }
}
